### PR TITLE
Spring3 migration

### PR DESCRIPTION
**Pull request by 07joshua03**
TODO:

- [x] Fix ConstraintViolationExpection (see test results)
- [ ] Remove trailing slashes everywhere and fix hrefs
- [x] Fix temporary removal of NonNull inside of Order 
- [x] Migrate Spring boot 3.1 etc.
- [x] See if other issues arise
- [ ]

### CODE REVIEW

**Comment by 07joshua03** [comment]
CodeQL will not work for now I think as per https://github.com/github/codeql-action/issues/1855

Leading slashes should still be fixed, as the current path (see WebConfiguration.java) is already deprecated: 'The method setUseTrailingSlashMatch(Boolean) from the type PathMatchConfigurer is deprecated since version 6.0'

Beyond that, I think the migration is done.

File: src/main/java/ch/wisv/events/WebConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.PathMatchConfigurer;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+public class WebConfiguration implements WebMvcConfigurer {
+
+    @Override
+    public void configurePathMatch(PathMatchConfigurer configurer) {
+      configurer.setUseTrailingSlashMatch(true);
```
Comment (JoepdeJong) [inline src/main/java/ch/wisv/events/WebConfiguration.java:12]: Seems like a good temporary fix. 

As described in https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#web-application-changes, we should manually define all paths with and without trailing slashes, or change the Servlet/Web filter. I am unable to find examples of the latter.

Lets stick to this temporary solution, but make a new issue describing this.

**Review by JoepdeJong — APPROVED** [review]
LGTM.

Reviewed all changes and tested locally.

Please make a new issue for `configurer.setUseTrailingSlashMatch(true);`

**Review by JoepdeJong — CHANGES_REQUESTED** [review]
I did not test the login with Connect, because its currently not running on my localhost. Can you make sure this still works?

**Comment by 07joshua03** [comment]
> I did not test the login with Connect, because its currently not running on my localhost. Can you make sure this still works?

I tested it with connect, and this still works as expected

File: src/main/java/ch/wisv/events/WebConfiguration.java (hunk 1 above)
Comment (07joshua03) [reply src/main/java/ch/wisv/events/WebConfiguration.java:12]: Created a new issue for this.

**Pull request merged**